evap/evaluation/models.py

```diff
@@ -2059,17 +2061,15 @@ def send_to_users_in_evaluations(self, evaluations, recipient_groups, use_cc, re
 
     def send_to_user(self, user, subject_params, body_params, use_cc, additional_cc_users=(), request=None):
         if not user.email:
-            warning_message = (
-                f"{user.full_name_with_additional_info} has no email address defined. Could not send email."
-            )
+            message = f"{user.full_name_with_additional_info} has no email address defined. Could not send email."
+            level = logging.ERROR
             # If this method is triggered by a cronjob changing evaluation states, the request is None.
             # In this case warnings should be sent to the admins via email (configured in the settings for logger.error).
             # If a request exists, the page is displayed in the browser and the message can be shown on the page (messages.warning).
             if request is not None:
-                logger.warning(warning_message)
-                messages.warning(request, _(warning_message))
-            else:
-                logger.error(warning_message)
+                level = logging.WARNING
+                messages.warning(request, _(message))
```

Comment:
Does `_(message)` even work with the user data already formatted into `message`? Don't we usually do `_("look at this data: {}").format(the_data)`?

Reply:
reverted the 'simplification' and fixed the issue. Only did it in the first place to keep under ruff's `too-many-branches`. This is now only covered with pylint

Reply:
Do you know why ruff raised a pylint-based lint while pylint itself did not raise it? If the two check the same, we should probably prefer ruff for speed, right?

Reply:
There was disagreement between pylint and ruff about how many branches there really are in the function. I think ruff counted one more than pylint, so the pylint check doesn't trigger while the ruff check does. Don't know if it's a ruff bug or if ruff thinks it's a pylint bug.

In general, I'd agree with @niklasmohrin to go with ruff for speed.

Reply:
Okay, maybe lets add this as a comment so we can revisit this later

Reply:
Ah well no. ruff has a bug there. It does not count branches inside `with` statements... [astral-sh/ruff#11313](https://www.github.com/astral-sh/ruff/issues/11313)